### PR TITLE
Rename Curve/Curve2D/Curve3D/Gradient `interpolate()` to `sample()`

### PR DESCRIPTION
**Pull request by Calinou**
"sampling" is a more accurate term than "interpolating" for what's happening when using that function.

See https://github.com/godotengine/godot/issues/54161#issuecomment-1193189694.

### CODE REVIEW

**Comment by akien-mga** [comment]
We discussed this in a PR meeting. It seems slightly better, and has some user support, but we're not sure it's _really_ going to make the API clearer, and it might annoy some users that the compat is broken here too.

But well, we can merge and see what happens. Needs a rebase.

File: editor/project_converter_3_to_4.cpp [hunk 1]
```diff
@@ -398,6 +398,7 @@ static const char *gdscript_function_renames[][2] = {
 	{ "import_animation_from_other_importer", "_import_animation" }, //EditorSceneFormatImporter
 	{ "import_scene_from_other_importer", "_import_scene" }, //EditorSceneFormatImporter
 	{ "instance_set_surface_material", "instance_set_surface_override_material" }, // RenderingServer
+	{ "interpolate", "sample" }, // Curve, Curve2D, Curve3D, Gradient
```
Comment (akien-mga) [inline editor/project_converter_3_to_4.cpp:401]: Aren't there other `interpolate` methods that will be wrongly changed?

Comment (Mickeon) [reply editor/project_converter_3_to_4.cpp:401]: There's **GPUParticles2D/3D**.`interpolate()`. There's also **Variant**.`interpolate()` but I don't believe it is exposed anywhere...

Comment (Calinou) [reply editor/project_converter_3_to_4.cpp:401]: > There's GPUParticles2D/3D.interpolate()

This property only exists in 4.0, so it should be fine.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**